### PR TITLE
GLPFinder - New ranking using detected effect

### PR DESCRIPTION
**Pull request by NicolasMatrices-v2**
- Leverage structure of pvalueCalc to include detected effect as part of its output.
- The difference between detected effect and true effect is positively correlated with power curve centeredness and shape.
- If we combine this value with the existing metrics (i..e MDE) we can probably improve our ranking algorithm.

### CODE REVIEW

File: R/GeoLift.R [hunk 1]
```diff
@@ -484,101 +495,116 @@ fn_treatment <- function(df,
 #'
 #' @export
 pvalueCalc <- function(data,
```
Comment (NicolasMatrices-v2) [inline R/GeoLift.R:497]: Changes start here.

File: R/GeoLift.R [hunk 2]
```diff
@@ -484,101 +495,116 @@ fn_treatment <- function(df,
...
-    PowerCalc <- augsynth::augsynth(fmla,
-                                    unit = location,
-                                    time = time,
-                                    data = data_aux,
-                                    t_int = treatment_start_time,
-                                    progfunc = "GSYN",
-                                    scm = T,
-                                    fixedeff = fixed_effects)
-  }
+  ave_treatment_convs <- sum(ascm_obj$data$y[which(ascm_obj$data$trt == 1), ]) / sum(ascm_obj$data$trt == 1)
```
Comment (NicolasMatrices-v2) [inline R/GeoLift.R:559]: This is a change.

File: R/GeoLift.R [hunk 3]
```diff
@@ -484,101 +495,116 @@ fn_treatment <- function(df,
...
 
-  return (
-    c(paste(locations, collapse=", "),
-      pVal,
-      tp,
-      es,
-      treatment_start_time,
-      investment,
-      ScaledL2Imbalance)
+  return(
```
Comment (NicolasMatrices-v2) [inline R/GeoLift.R:596]: this is a change.

File: R/GeoLift.R [hunk 4]
```diff
@@ -1805,149 +1846,165 @@ GeoLiftPowerFinder <- function(data,
...
 
-          for (i in 1:ncol(a)) {
-            results <- rbind(results, data.frame(location = a[[1,i]],
-                                                 pvalue = as.numeric(a[[2,i]]),
-                                                 duration = as.numeric(a[[3,i]]),
-                                                 lift = as.numeric(a[[4,i]]),
-                                                 treatment_start = as.numeric(a[[5,i]]),
-                                                 ScaledL2Imbalance = as.numeric(a[[7,i]]) ) )
+          for (i in 1:ncol(simulation_results)) {
+            results <- rbind(results, data.frame(
```
Comment (NicolasMatrices-v2) [inline R/GeoLift.R:1933]: This is a change.

File: R/GeoLift.R [hunk 5]
```diff
@@ -1805,149 +1846,165 @@ GeoLiftPowerFinder <- function(data,
...
-            results <- rbind(results, data.frame(location=aux[1],
-                                                 pvalue = as.numeric(aux[2]),
-                                                 duration = as.numeric(aux[3]),
-                                                 lift = as.numeric(aux[4]),
-                                                 treatment_start = as.numeric(aux[5]),
-                                                 ScaledL2Imbalance = as.numeric(aux[7]) ) )
+              stat_func = stat_func
+            ))
+
+            results <- rbind(results, data.frame(
```
Comment (NicolasMatrices-v2) [inline R/GeoLift.R:1963]: this is a change.

**Review by ArturoEsquerra — APPROVED** [review]

**Pull request merged**